### ISSUE TITLE
Transition component expected behavior? (Multiple elements created when toggling fast)

### ISSUE THREAD

**Issue by johot** [drcmda/react-spring#211]
So I wanted to use the `<Transition/>` component for fading a spinner in/out as it mounts and unmounts. I looked at this sample from the docs for my [implementation](https://codesandbox.io/s/n329on8q1p):

```jsx
<Transition from={{ opacity: 0 }} enter={{ opacity: 1 }} leave={{ opacity: 0 }}>
    {visible && (styles => <div style={styles}>Single Component</div>)}
</Transition>
```

When toggling the `visible` flag fast I will see another component get created while the original spinner is fading out.

Is this the expected behavior? I could understand if it is since it might see the first spinner instance as "unmounted" thus needing to mount another one? But couldn't it somehow reuse the first component as long as the instance is not fully animated out. Or am I missing something?

I prepared a code sandbox sample that should make it easy to test this, using latest `react` and `react-spring` versions. Also made sure to cache the inner component to see if that made any changes.

Take a look at the sample code and sample here:

https://codesandbox.io/s/n329on8q1p

Thanks for a great library!

**Comment by drcmda** [comment]
Yep, that's expected. Transition guarantees exit transitions, even if the same element, with the same key, comes in again. In the transition component you're always dealing with a stack, therefore it's often better to position items absolutely, unless you're working with lists where items *should* push others aways. https://codesandbox.io/embed/oxml10qxlz

**Comment by johot** [comment]
Thanks for clarifying @drcmda !

So for example you mean that a transition where something goes in from left, stops in the middle then goes out right would look strange if the component is marked as visible during the "leave transition" since it would jank back to the left "entering position"?

In my use case would it be better to use another primitive, `<Spring />` etc?

Also I think it would be nice to maybe have a note about this in the docs? What do you think?

Thanks again!

**Comment by drcmda** [comment]
Yes, i would use spring for on/off kind of animations, anything that's going to toggle. It's often better (faster) to keep elements in the render tree even if they're invisible instead of re-mounting them conditionally, even for things like sidebars and such. A transition is more suitable for things that come in and out and that should definitively disappear when done - i'd put a spinner probably into a transition. But transitions are always stacks due to the exiting elements.

**Comment by johot** [comment]
Thanks @drcmda great reply. I'll see if I can do some clarifications in the Readme for this if I have time to do that PR :)
